runtime/bindings/python/iree/runtime/array_interop.py

```diff
@@ -216,8 +216,8 @@ def asdevicearray(
         logging.warn(
             "Implicit dtype conversion of a DeviceArray forces a host transfer"
         )
-    # First get an ndarray.
-    a = np.asarray(a, dtype=dtype)
+    # First get an ndarray. Needs to be C-contiguous, enforcing it here.
+    a = np.asarray(a, dtype=dtype, order="C")
```

Comment:
This may introduce a copy (https://numpy.org/doc/stable/reference/generated/numpy.asarray.html), but the comment above says that is okay for this API.

> This function may make a defensive copy or cause implicit
    transfers to satisfy the request. If this is important to you, then a lower
    level API is likely more appropriate.